node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+        inputs
+            .into_iter()
+            .for_each(|(error, expected_msg)| assert_eq!(error.to_string(), expected_msg));
+        assert_eq!(inputs_count, PaymentAdjusterError::VARIANT_COUNT + 2)
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "internal error: entered unreachable code: This error contains no \
+    specifications"
+    )]
+    fn error_message_for_input_referring_to_no_issues_cannot_be_made() {
+        let _ = PaymentAdjusterError::EarlyNotEnoughFeeForSingleTransaction {
+            number_of_accounts: 0,
+            transaction_fee_opt: None,
+            service_fee_opt: None,
+        }
+        .to_string();
+    }
+
+    #[test]
+    fn we_can_say_if_error_occurred_after_insolvency_was_detected() {
+        let inputs = vec![
+            PaymentAdjusterError::AllAccountsEliminated,
+            PaymentAdjusterError::EarlyNotEnoughFeeForSingleTransaction {
+                number_of_accounts: 0,
+                transaction_fee_opt: Some(TransactionFeeImmoderateInsufficiency {
+                    per_transaction_requirement_minor: 0,
+                    cw_transaction_fee_balance_minor: Default::default(),
+                }),
+                service_fee_opt: None,
+            },
+            PaymentAdjusterError::EarlyNotEnoughFeeForSingleTransaction {
+                number_of_accounts: 0,
+                transaction_fee_opt: None,
+                service_fee_opt: Some(ServiceFeeImmoderateInsufficiency {
+                    total_service_fee_required_minor: 0,
+                    cw_service_fee_balance_minor: 0,
+                }),
+            },
+            PaymentAdjusterError::EarlyNotEnoughFeeForSingleTransaction {
+                number_of_accounts: 0,
+                transaction_fee_opt: Some(TransactionFeeImmoderateInsufficiency {
+                    per_transaction_requirement_minor: 0,
+                    cw_transaction_fee_balance_minor: Default::default(),
+                }),
+                service_fee_opt: Some(ServiceFeeImmoderateInsufficiency {
+                    total_service_fee_required_minor: 0,
+                    cw_service_fee_balance_minor: 0,
+                }),
+            },
+            PaymentAdjusterError::LateNotEnoughFeeForSingleTransaction {
+                original_number_of_accounts: 0,
+                number_of_accounts: 0,
+                original_service_fee_required_total_minor: 0,
+                cw_service_fee_balance_minor: 0,
+            },
+        ];
+        let inputs_count = inputs.len();
+        let results = inputs
+            .into_iter()
+            .map(|err| err.insolvency_detected())
+            .collect::<Vec<_>>();
+        assert_eq!(results, vec![true, true, true, true, true]);
+        assert_eq!(inputs_count, PaymentAdjusterError::VARIANT_COUNT + 2)
+    }
+
+    #[test]
+    fn tinier_but_larger_in_weight_account_is_prioritized_and_gains_up_to_its_disqualification_limit(
```

Comment:
We want to discuss the name change or the test removal.

In case name change is picked, then a suggestion: `account_with_larger_weight_is_prioritized_and_gains_up_to_its_disqualification_limit()`.